modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/SyncmlCredentials.java

```diff
@@ -42,15 +42,14 @@ public String generateCredData(String nextNonce) throws UnsupportedEncodingExcep
             String usrPwdNonce = usrPwdHash + ":" + nonce;
             usrPwdNonceHash = Base64.encodeBase64String(digest.digest(usrPwdNonce.getBytes(Constants.UTF_8)));
         } catch (UnsupportedEncodingException e) {
-            throw new UnsupportedEncodingException("Problem occurred in encoding credentials data.");
+            throw new SyncmlMessageFormatException("Problem occurred in encoding credentials data.", e);
         } catch (NoSuchAlgorithmException e) {
-            throw new NoSuchAlgorithmException("Problem occurred in generating password hash.", e);
+            throw new SyncmlMessageFormatException("Problem occurred in generating password hash.", e);
         }
         return usrPwdNonceHash;
     }
 
-    public String generateRST(String username, String password) throws UnsupportedEncodingException,
-            NoSuchAlgorithmException {
+    public String generateRST(String username, String password) throws SyncmlMessageFormatException {
```

Comment:
do not see any instant variables in this Class definition. Thus, there is no need as it seems to instantiate this class. Better to make this a utility class and methods static.